idx/cassandra/config.go

```diff
@@ -0,0 +1,120 @@
+package cassandra
+
+import (
+	"errors"
+	"flag"
+	"time"
+
+	"github.com/rakyll/globalconf"
+	log "github.com/sirupsen/logrus"
+)
+
+// time units accepted by time.ParseDuration
+const timeUnits = "Valid time units are 'ns', 'us' (or 'µs'), 'ms', 's', 'm', 'h'"
+
+// CliConfig is a cassandra IdxConfig. It is instantiated with default values which can then be changed.
+var CliConfig = NewIdxConfig()
+
+// IdxConfig stores configuration settings for a cassandra index
+type IdxConfig struct {
+	Enabled          bool
+	pruneInterval    time.Duration
+	updateCassIdx    bool
+	updateInterval   time.Duration
+	updateInterval32 uint32
+
+	writeQueueSize int
+
+	ssl                      bool
+	auth                     bool
+	hostverification         bool
+	createKeyspace           bool
+	schemaFile               string
+	keyspace                 string
+	hosts                    string
+	capath                   string
+	username                 string
+	password                 string
+	consistency              string
+	timeout                  time.Duration
+	numConns                 int
+	protoVer                 int
+	disableInitialHostLookup bool
+}
+
+// NewIdxConfig returns IdxConfig with default values set.
+func NewIdxConfig() *IdxConfig {
+	return &IdxConfig{
+		Enabled:                  true,
+		hosts:                    "localhost:9042",
+		keyspace:                 "metrictank",
+		consistency:              "one",
+		timeout:                  time.Second,
+		numConns:                 10,
+		writeQueueSize:           100000,
+		updateCassIdx:            true,
+		updateInterval:           time.Hour * 3,
+		pruneInterval:            time.Hour * 3,
+		protoVer:                 4,
+		createKeyspace:           true,
+		schemaFile:               "/etc/metrictank/schema-idx-cassandra.toml",
+		disableInitialHostLookup: false,
+		ssl:                      false,
+		capath:                   "/etc/metrictank/ca.pem",
+		hostverification:         true,
+		auth:                     false,
+		username:                 "cassandra",
+		password:                 "cassandra",
+	}
+}
+
+// Validate validates IdxConfig settings
+func (cfg *IdxConfig) Validate() error {
+	if cfg.updateInterval == 0 {
+		return errors.New("updateInterval must be greater than 0. " + timeUnits)
+	}
+	if cfg.pruneInterval == 0 {
+		return errors.New("pruneInterval must be greater then 0. " + timeUnits)
+	}
+	if cfg.timeout == 0 {
+		return errors.New("timeout must be greater than 0. " + timeUnits)
+	}
```

Comment:
all these errors are be misleading when a user entered a non-zero value that didn't parse.
Perhaps reword as "timeout invalid. Must be greater than 0 and parseable with https://golang.org/pkg/time/#ParseDuration" or something

Reply:
since we use flag.ExitOnError, that should work in theory, but in practice, it doesn't seem to
when I use `update-interval = 4H` I get:
```
metrictank_1_cced215491fa | 2018-11-21 05:58:48.740 [FATAL] cassandra-idx: Config validation error. updateInterval must be greater than 0. Valid time units are 'ns', 'us' (or 'µs'), 'ms', 's', 'm', 'h'
```
meaning we don't detect parse errors, and just pass the 0 value through.
unfortunately, we already knew this as brought up in https://github.com/grafana/metrictank/issues/944#issuecomment-398356914

Reply:
why are we still using `github.com/rakyll/globalconf`?   we have our own fixed version `github.com/grafana/globalconf`